src/tree/containers/ContainersItem.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { KnownActiveRevisionsMode, type Container, type Revision } from "@azure/arm-appcontainers";
+import { nonNullValue, nonNullValueAndProp, type TreeElementBase } from "@microsoft/vscode-azext-utils";
+import { type AzureSubscription, type ViewPropertiesModel } from "@microsoft/vscode-azureresources-api";
+import * as deepEqual from 'deep-eql';
+import { TreeItemCollapsibleState, type TreeItem } from "vscode";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../utils/localize";
+import { getParentResource } from "../../utils/revisionDraftUtils";
+import { treeUtils } from "../../utils/treeUtils";
+import { type ContainerAppModel } from "../ContainerAppItem";
+import { RevisionDraftDescendantBase } from "../revisionManagement/RevisionDraftDescendantBase";
+import { RevisionDraftItem } from "../revisionManagement/RevisionDraftItem";
+import { ContainerItem } from "./ContainerItem";
+import { EnvironmentVariablesItem } from "./EnvironmentVariablesItem";
+import { ImagesItem } from "./ImagesItem";
+
+export class ContainersItem extends RevisionDraftDescendantBase {
+    id: string;
+    label: string;
+    private containers: Container[] = [];
+
+    constructor(public readonly subscription: AzureSubscription,
+        public readonly containerApp: ContainerAppModel,
+        public readonly revision: Revision,) {
+        super(subscription, containerApp, revision);
+        this.id = `${containerApp.id}/containers`;
```

Comment:
This id may not always be unique, for example in multiple revisions mode.

You can probably fix this by doing something similar to what I did in the ID for `ScaleRuleItem`!